File: FruitWholesale.Client/src/app/features/dashboard/dashboard.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { DashboardCharts, DashboardPeriod, DashboardSummary, SalesVsPurchases, TrendPoint } from '../../core/models/dashboard.model';

@Injectable({ providedIn: 'root' })
export class DashboardService {
  private readonly baseUrl = `${environment.apiUrl}/dashboard`;

  constructor(private readonly http: HttpClient) {}

  getSummary(): Observable<DashboardSummary> {
    return this.http.get<DashboardSummary>(`${this.baseUrl}/summary`);
  }

  getCharts(): Observable<DashboardCharts> {
    return this.http.get<DashboardCharts>(`${this.baseUrl}/charts`);
  }

  getSalesTrend(period: DashboardPeriod): Observable<TrendPoint[]> {
    return this.http.get<TrendPoint[]>(`${this.baseUrl}/sales-trend`, { params: { period } });
  }

  getSalesVsPurchases(period: DashboardPeriod): Observable<SalesVsPurchases> {
    return this.http.get<SalesVsPurchases>(`${this.baseUrl}/sales-vs-purchases`, { params: { period } });
  }

  getCashTrend(): Observable<TrendPoint[]> {
    return this.http.get<TrendPoint[]>(`${this.baseUrl}/cash-trend`);
  }

  getProfitTrend(): Observable<TrendPoint[]> {
    return this.http.get<TrendPoint[]>(`${this.baseUrl}/profit-trend`);
  }
}

File: FruitWholesale.Mobile/lib/features/dashboard/dashboard_service.dart
import '../../core/api/api_client.dart';
import 'dashboard_models.dart';

class DashboardService {
  final ApiClient _api;

  DashboardService(this._api);

  Future<DashboardSummary> getSummary() async {
    final json = await _api.get('/dashboard/summary');
    return DashboardSummary.fromJson(json as Map<String, dynamic>);
  }

  Future<DashboardCharts> getCharts() async {
    final json = await _api.get('/dashboard/charts');
    return DashboardCharts.fromJson(json as Map<String, dynamic>);
  }

  Future<List<TrendPoint>> getSalesTrend(DashboardPeriod period) async {
    final json = await _api.get('/dashboard/sales-trend', query: {'period': period.toApi}) as List;
    return json.map((e) => TrendPoint.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<SalesVsPurchases> getSalesVsPurchases(DashboardPeriod period) async {
    final json = await _api.get('/dashboard/sales-vs-purchases', query: {'period': period.toApi});
    return SalesVsPurchases.fromJson(json as Map<String, dynamic>);
  }

  Future<List<TrendPoint>> getCashTrend() async {
    final json = await _api.get('/dashboard/cash-trend') as List;
    return json.map((e) => TrendPoint.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<List<TrendPoint>> getProfitTrend() async {
    final json = await _api.get('/dashboard/profit-trend') as List;
    return json.map((e) => TrendPoint.fromJson(e as Map<String, dynamic>)).toList();
  }
}
